upgrade to Julia v0.7

RalphAS opened the pull request:
This works on a system with patched FileIO and ImageFiltering, so don't look too hard until those are updated and tagged.

timholy commented:
Thanks! I worry a bit about conflicts with #123, also a very big patch. @SimonDanisch, care to chime in with your thoughts?

RalphAS commented:
I naively thought that "binarybuilder" indicated a narrowly focussed update which would be easy to merge.  After looking closer, I learned otherwise.

Accordingly I have made a new branch (on my fork, not currently a PR here), standing on Simon's shoulders.  The result runs (and passes tests!) with local patches but not on the CI servers. (Some published but untagged revisions of dependencies might suffice, but I gather that it is bad form to use those in CI scripts).   Should I force-replace this PR branch with the one that incorporates Simon's commits?
@timholy @SimonDanisch

bjarthur commented:
please tag a release before dropping support for julia 0.6!

bjarthur commented:
PR in metadata now...  https://github.com/JuliaLang/METADATA.jl/pull/15632

RalphAS commented:
I have rebased (and added a few new patches) to accommodate the merge of #121.  This is still awaiting merging of updates to FileIO and ImageFiltering.

musm commented on .travis.yml:
```diff
@@ -3,18 +3,21 @@ os:
   - nightly
 matrix:
   allow_failures:
+    - julia: 1.0
```
probably remove this

RalphAS commented:
Subsumed in #125
